src/index.js

```diff
@@ -16,12 +16,24 @@ const pipeline = (...functions) => {
   }
 }
 
+const head = arr => {
+  return arr[0]
+}
+
+const tail = arr => {
+  return arr.slice(1)
+}
+
 const retrieveData = element => {
   return element.value;
 }
 
-const makeIdea = (title, desc) => {
-  return {title, desc};
+const addDescription = idea => {
+  return {...idea, desc: retrieveData(descInput)}
```

Comment:
Yeah, I understand how using retrieveData with the input element is a side effect. I'm going to breakout the DOM parts into their own file so all the side effects happen in one place.